src/components/providers/ConnectionStatusProvider.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { PropsWithChildren, useState } from 'react'
+import ConnectionStatusContext from '../../contexts/connection-status-context'
+import { ConnectionStatus } from '../../enums/connection-status'
+import { ConnectionStatusContextType } from '../../types/contexts'
+import { LocalStorageKey } from '../../enums/local-storage-key'
+import LocalStorageService from '../../services/local-storage-service'
+import { useNavigate } from 'react-router'
+
+export const ConnectionStatusProvider = ({ children }: PropsWithChildren<any>) => {
+  const localStorageService = new LocalStorageService()
+  const navigate = useNavigate()
+  const requestUrl: string = process.env.REACT_APP_PUBLIC_BACKEND_API || 'http://localhost:3000'
+  const pollingDelay: string | number = process.env.ONLINE_POLLING_DELAY || 20000
+  // eslint-disable-next-line no-undef
+  let intervalId: NodeJS.Timer
+
+  const getInitialOnlineInfoBannerVisibility = () => {
+    const savedOnlineInfoBannerVisibility = localStorageService.getItem(LocalStorageKey.IsOnlineBannerSeen)
+    if (!savedOnlineInfoBannerVisibility) {
+      return false
+    }
+    return JSON.parse(savedOnlineInfoBannerVisibility) === true
+  }
+
+  const [isOnlineInfoBannerVisible, setIsOnlineInfoBannerVisible] = useState<boolean>(getInitialOnlineInfoBannerVisibility)
+
+  const updateOnlineInfoBannerVisibility = (isVisible: boolean) => {
+    localStorageService.addItem(LocalStorageKey.IsOnlineBannerSeen, String(isVisible))
+    setIsOnlineInfoBannerVisible(isVisible)
+  }
+
+  const resetOnlineInfoBanner = () => {
+    localStorageService.removeItem(LocalStorageKey.IsOnlineBannerSeen)
+  }
+
+  const getInitialGoOnlineButtonState = () => {
+    const savedGoOnlineButtonState = localStorageService.getItem(LocalStorageKey.IsGoOnlineButtonVisible)
+    if (!savedGoOnlineButtonState) {
+      return false
+    }
+
+    return JSON.parse(savedGoOnlineButtonState) === true
+  }
+
+  const [showGoOnlineButton, setShowGoOnlineButton] = useState<boolean>(getInitialGoOnlineButtonState)
+
+  const updateGoOnlineButtonVisibility = (isVisible: boolean) => {
+    localStorageService.addItem(LocalStorageKey.IsGoOnlineButtonVisible, String(isVisible))
+    setShowGoOnlineButton(isVisible)
+  }
+
+  const checkIfApplicationIsOnline = async () : Promise<void> => {
+    const request = await fetch(requestUrl + '/heartbeat')
+    if (request.status === 200) {
+      clearInterval(intervalId)
+      updateGoOnlineButtonVisibility(true)
+      updateOnlineInfoBannerVisibility(true)
+    }
+  }
+
+  const initPolling = () => {
+    intervalId = setInterval(checkIfApplicationIsOnline, +pollingDelay)
```

Comment:
I would use `parseInt` above where we read the env variable instead of using the `+` operator here.

Reply:
done